artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/ServerSessionPacketHandler.java

```diff
@@ -635,6 +648,27 @@ private void slowPacketHandler(final Packet packet) {
                   }
                   break;
                }
+               case CREATE_PRODUCER: {
+                  CreateProducerMessage message = (CreateProducerMessage) packet;
+                  // this is used to create/destroy the producer so needs to be unique
+                  String senderName = null;
+                  if (!producers.containsKey(message.getId())) {
+                     senderName = PRODUCER_ID_PREFIX + UUIDGenerator.getInstance().generateUUID();
+                     producers.put(message.getId(), senderName);
+                  }
+                  session.addProducer(senderName, ActiveMQClient.DEFAULT_CORE_PROTOCOL, message.getAddress() != null ? message.getAddress().toString() : null);
+                  break;
+               }
+               case REMOVE_PRODUCER: {
+                  RemoveProducerMessage message = (RemoveProducerMessage) packet;
+                  String remove = producers.remove(message.getId());
+                  if (remove != null) {
+                     session.removeProducer(remove);
+                  } else {
+                     ActiveMQServerLogger.LOGGER.producerDoesNotExist(session.getName());
```

Comment:
Should probably say which producer (id) didnt exist to give some detail beyond the session. Can you tell which connection it is from the session name? If not then perhaps that also to aid tracking down problem client.